graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/FederationDirectives.java

```diff
@@ -28,136 +29,89 @@
             .build();
-
-    /* directive @key(fields: _FieldSet!) on OBJECT | INTERFACE */
-
-    public static final String keyName = "key";
-
     public static final GraphQLDirective key = newDirective()
             .name(keyName)
             .validLocations(OBJECT, INTERFACE)
             .argument(fieldsArgument)
             .build();
-
-    public static final GraphQLDirective key(String fields) {
-        return newDirective(key)
-                .argument(fieldsArgument(fields))
-                .build();
-    }
-
-    public static final DirectiveDefinition keyDefinition = newDirectiveDefinition()
-            .name(keyName)
-            .directiveLocations(Arrays.asList(DL_OBJECT, DL_INTERFACE))
-            .inputValueDefinition(fieldsDefinition)
-            .build();
-
-    /* directive @external on FIELD_DEFINITION */
-
-    public static final String externalName = "external";
-
-    public static final GraphQLDirective external = newDirective()
-            .name(externalName)
-            .validLocations(FIELD_DEFINITION)
-            .build();
-
-    public static final DirectiveDefinition externalDefinition = newDirectiveDefinition()
-            .name(externalName)
-            .directiveLocations(Arrays.asList(DL_FIELD_DEFINITION))
-            .build();
-
-    /* directive @requires(fields: _FieldSet!) on FIELD_DEFINITION */
-
-    public static final String requiresName = "requires";
-
     public static final GraphQLDirective requires = newDirective()
             .name(requiresName)
             .validLocations(FIELD_DEFINITION)
             .argument(fieldsArgument)
             .build();
 
-    public static final GraphQLDirective requires(String fields) {
-        return newDirective(requires)
-                .argument(fieldsArgument(fields))
-                .build();
-    }
-
-    public static final DirectiveDefinition requiresDefinition = newDirectiveDefinition()
-            .name(requiresName)
-            .directiveLocations(Arrays.asList(DL_FIELD_DEFINITION))
-            .inputValueDefinition(fieldsDefinition)
-            .build();
-
     /* directive @provides(fields: _FieldSet!) on FIELD_DEFINITION */
-
-    public static final String providesName = "provides";
-
     public static final GraphQLDirective provides = newDirective()
             .name(providesName)
             .validLocations(FIELD_DEFINITION)
             .argument(fieldsArgument)
             .build();
-
-    public static final GraphQLDirective provides(String fields) {
-        return newDirective(provides)
-                .argument(fieldsArgument(fields))
-                .build();
-    }
-
-    public static final DirectiveDefinition providesDefinition = newDirectiveDefinition()
-            .name(providesName)
+    private static final InputValueDefinition fieldsDefinition = newInputValueDefinition()
+            .name("fields")
+            .type(new NonNullType(new TypeName(_FieldSet.typeName)))
```

Comment:
Seems like we might be able to reuse this type and be more consistent over TypeReference, TypeName, or the actual type.

Reply:
Reuse it where?

Reply:
Seems like it's gone